[EDGE CIE-5316] c8yedge tool documentation -- DON'T MERGE OR DELETE (#3329)
SoftwareAG/c8y-docs: content/edge-kubernetes/installing-edge-on-k8.md

  - edge_server
---

Edge can be installed using one of two supported methods. The method you choose depends on whether you already have a Kubernetes environment or would like the tool to provision one for you.
  - **[Installing with c8yedge tool](#install-edge-with-c8yedge-cli) (Recommended for first-time or simplified setups)**
    <br/>
    This method is ideal if you **do not already have a Kubernetes cluster** and want a simplified, self-contained setup.
    <br/>
    The `c8yedge` command-line tool automates the entire process of preparing the environment and installing Edge. Specifically, it:
    - Installs [K3s](https://docs.k3s.io/installation), a lightweight Kubernetes distribution optimized for edge environments.
    - Deploys the Edge Operator and uses it to install {{< product-c8y-iot >}} Edge with the below configuration
      - **Name**: `c8yedge`
      - **Company**: `Edge Bootstrap`
      - **Domain**: `edgebootstrap.example`

>       - **Email**: `company@edgebootstrap.example`

Reviewer: So as not to alarm users away from using `c8yedge`, we should reassure them that these can all be changed later.

Reviewer 2: Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/210b2f6fd26554888d6f68f430c46b0eb6023b60

    <br/>
    This is the recommended method for users who want a quick, consistent setup experience with minimal Kubernetes knowledge.
    <br/>
    <br/>

  - **[Installing with the Edge Operator](#install-edge-with-edge-operator) (For users with an existing Kubernetes setup)**
    <br/>

>     This method is suitable for users who already have a Kubernetes cluster (typically with a single worker node) and want to install Edge using the Kubernetes-native approach.

Reviewer: ```suggestion
    This method is suitable for users who already have a Kubernetes cluster and want to install Edge using the Kubernetes-native approach. Even a single-node Kubernetes cluster will suffice, but is not required.
```
as a power user, I don't want to be told how I have to set up _my_ kubernetes cluster :-)

Reviewer 2: Resolved: https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/4cef6481dd01c65061cfa90b9453a024de4a857b

    <br/>
    In this case, you are expected to:

>     - Set up and manage the Kubernetes cluster yourself (including K3s or any compatible Kubernetes distribution).

Reviewer: ```suggestion
    - Set up and manage the Kubernetes cluster yourself (using K3s or any compatible Kubernetes distribution).
```

Reviewer 2: Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/16583477b55aa3a6c72651316e9a2fc282583fc1

>     - Install any [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites)

Reviewer: ```suggestion
    - Do any system configuration or software installation required by the [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites)
```
(a lot of those requirements are non-software)

Reviewer 2: Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/485b151d04a47a58ffc71043a0a1fe908e2789bf
